packages/arb-token-bridge-ui/src/hooks/useArbQueryParams.tsx

```diff
@@ -43,18 +43,28 @@ export const useArbQueryParams = () => {
   })
 }
 
+const isMax = (amount: string | null | undefined) =>
+  amount?.toLowerCase() === AmountQueryParamEnum.MAX
+
 // Our custom query param type for Amount field - will be parsed and returned as a string,
 // but we need to make sure that only valid numeric-string values are considered, else return '0'
 // Defined here so that components can directly rely on this for clean amount values and not rewrite parsing logic everywhere it gets used
 const AmountQueryParam = {
   // type of amount is always string | undefined coming from the input element onChange event `e.target.value`
-  encode: (amount: string | null | undefined) => amount,
+  encode: (amount: string | null | undefined) => {
+    const amountRegex = new RegExp(/^\d*(\.\d*)?$/)
+    if (isMax(amount)) {
+      return amount
+    }
+    return amountRegex.test(amount ?? '') ? amount : ''
+  },
   decode: (amount: string | (string | null)[] | null | undefined) => {
     const amountStr = amount?.toString()
 
     // to catch random string like `amount=asdf` from the URL
-    if (isNaN(Number(amountStr))) {
-      if (amountStr?.toLowerCase() === AmountQueryParamEnum.MAX) {
+    // to catch negative number
+    if (isNaN(Number(amountStr)) || Number(amountStr) < 0) {
+      if (isMax(amountStr)) {
```

Comment:
Can it be `max` in this if? In the line above, we check that the number is either `NaN` or negative.

Reply:
yea, `max` is a string which will return true from `isNaN`

Reply:
i guess we can move it out to before this check